grpc/src/main/java/io/stargate/grpc/service/BatchHandler.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.grpc.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import io.grpc.stub.StreamObserver;
+import io.stargate.db.Persistence;
+import io.stargate.db.Persistence.Connection;
+import io.stargate.db.Result.Prepared;
+import io.stargate.grpc.service.Service.PrepareInfo;
+import io.stargate.proto.QueryOuterClass.Response;
+import java.util.concurrent.CompletionStage;
+
+class BatchHandler extends MessageHandler {
```

Comment:
Probably there will be even more common parts between `BatchHandler` and `QueryHandler`.

Side note: The one problem that I see is that `QueryParameters` and `BatchParameters` don't share the same interface nor inherit from the same pattern (as far as I know, it is not possible in protobuf).
We could reduce even more duplication if that would be possible.